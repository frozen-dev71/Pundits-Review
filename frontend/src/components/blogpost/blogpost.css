/*
// ======================================
// CSS for Blog Post Component 
// ======================================
*/


.blog-post-wrapper {
	display: flex;
	justify-content: center;
	flex-direction: column;
	align-items: center;
}


.blog-loading-container {
	display: flex;
	justify-content: center;
	flex-direction: column;
	margin-top: 15vh;
	align-items: center;
}