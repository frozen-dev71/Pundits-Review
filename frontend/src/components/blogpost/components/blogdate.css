/*
// ======================================
// CSS for Blog Date Component 
// ======================================
*/

.blog-date-wrapper {
    padding: 2em 0;
}

.blog-date-text {
	font-size: .75em;
	letter-spacing: .2em;
	text-align: center;
}


/*
// ======================================
// MEDIA QUERIES
// ======================================
*/



@media (min-width: 750px) {

	.blog-date-text {
		
		font-size: 1em;

	}
}



@media (min-width: 1200px) {

	.blog-date-text {
		
		font-size: 1.5em;

	}


	.blog-date-wrapper {
	    padding: 4em 0;
	}
}