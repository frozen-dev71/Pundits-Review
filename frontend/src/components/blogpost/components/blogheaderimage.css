/*
// ======================================
// CSS for Blog Header Image
// ======================================
*/


.blog-header-image {
	width: 60vw;
    height: 50vh;
    background-position: top;
    background-repeat: no-repeat;
    background-size: cover;
    border: solid black 5px;
}




/*
// ======================================
// MEDIA QUERIES
// ======================================
*/



@media (min-width: 750px) {

	.blog-header-image {
		
		width: 50vw;

	}
}



@media (min-width: 1000px) {

	.blog-header-image {
		
		width: 40vw;

	}
}



@media (min-width: 1400px) {

	.blog-header-image {
		
		width: 30vw;
    	height: 60vh;

	}
}