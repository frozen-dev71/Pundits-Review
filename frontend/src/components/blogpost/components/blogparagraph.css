/*
// ======================================
// CSS for Blog Paragraph Component
// ======================================
*/


.blog-paragraph-text {
	padding: 2em 0;
    width: 80vw;
    font-size: .85em;
    line-height: 1.75em;
    letter-spacing: .025em;
}



/*
// ======================================
// MEDIA QUERIES
// ======================================
*/



@media (min-width: 650px) {

	.blog-paragraph-text {
		
		font-size: 1em;
		padding: 3em 0;

	}
}



@media (min-width: 750px) {

	.blog-paragraph-text {
		
		width: 70vw;

	}
}



@media (min-width: 900px) {

	.blog-paragraph-text {
		
		padding: 4em 0;

	}
}


@media (min-width: 1100px) {

	.blog-paragraph-text {
		
		font-size: 1.25em;
		width: 60vw;

	}
}



@media (min-width: 1400px) {

	.blog-paragraph-text {
		
		width: 50vw;

	}
}


@media (min-width: 1600px) {

	.blog-paragraph-text {
		
		width: 45vw;

	}
}



