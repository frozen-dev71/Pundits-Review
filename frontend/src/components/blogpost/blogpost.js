// ======================================
// Blog Post Component - Will wait for content to load then display the Blog Post Content Component 
// ======================================

// ===========
// IMPORTS 
// ===========

// General Imports 
import React, { Component } from "react";
import { render } from "react-dom";
import { connect } from "react-redux";
import { usePromiseTracker } from "react-promise-tracker";

// Local JS Imports
import { Loading } from './../global/components/loading';
import { BlogPostContent } from './components/blogpostcontent';

// CSS Imports 
import './blogpost.css';


// ===========
// Blog Post Component 
// ===========

export const BlogPost = props => {

	// Display loading screen until Promise resolved & blog post content has loaded
	const { promiseInProgress } = usePromiseTracker();

	let result;

	if (promiseInProgress === false) {

		result = (

				<BlogPostContent postName={props.postName}/>

			);


	} else {

		result = (

				<div className="blog-loading-container">

					<Loading />

				</div>

			);
	}


	return result;


}