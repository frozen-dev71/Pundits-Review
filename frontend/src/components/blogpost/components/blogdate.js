// ======================================
// Blog Date Component - Displays the Date of publication
// ======================================

// ===========
// IMPORTS 
// ===========

// General Imports 
import React, { Component } from "react";
import { render } from "react-dom";

// Local JS Imports


// CSS Imports 
import './blogdate.css';


// ===========
// Blog Header Image Component 
// ===========

export const BlogDate = props => {


	let result = (

			<div className="blog-date-wrapper">
				<p className="blog-date-text body-text-light">{props.date}</p>
			</div>

		);


	return result;

}