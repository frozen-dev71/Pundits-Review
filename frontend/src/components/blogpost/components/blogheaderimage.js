// ======================================
// Blog Header Image Component - Displays the Header Image for each blog post 
// ======================================

// ===========
// IMPORTS 
// ===========

// General Imports 
import React, { Component } from "react";
import { render } from "react-dom";

// Local JS Imports


// CSS Imports 
import './blogheaderimage.css';


// ===========
// Blog Header Image Component 
// ===========

export const BlogHeaderImage = props => {

	let imageStyle = {

		backgroundImage: "url(" + props.imageURL + ")"

	}

	let result = (

			<span style={imageStyle} className="blog-header-image"></span>

		);


	return result;

}