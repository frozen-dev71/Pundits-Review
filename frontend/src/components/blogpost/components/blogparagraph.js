// ======================================
// Blog Paragraph Component - Displays paragraph text as supplied by props
// ======================================

// ===========
// IMPORTS 
// ===========

// General Imports 
import React, { Component } from "react";
import { render } from "react-dom";

// Local JS Imports


// CSS Imports 
import './blogparagraph.css';


// ===========
// Blog Header Image Component 
// ===========

export const BlogParagraph = props => {


	let result;


	if (props.text !== '""') {

		result = (

			<p className="blog-paragraph-text body-text-light">{props.text}</p>

		);


	} else {

		result = <span></span>
	}


	return result;

}