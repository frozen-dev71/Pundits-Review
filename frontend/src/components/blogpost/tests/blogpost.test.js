// ======================================
// This file tests for the components listed in the blogpost sub-directory
// ======================================


// ===========
// TEST COVERAGE
// ===========

// 1. Check that BlogPost renders correctly
// 2. Check that BlogDate renders correctly
// 3. Check that BlogHeaderImage renders correctly
// 4. Check that BlogParagraph renders correctly
// 5. Check that BlogPlayerCard renders correctly
// 6. Check that BlogPostContent renders correctly
// 7. Check that BlogPostHeadline renders correctly
// 8. Check that BlogPullQuote renders correctly
// 9. Check that BlogSnippets renders correctly

// ===========
// IMPORTS 
// ===========


// General Imports
import React from 'react';
import Enzyme, { shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from "react-redux";
import toJson from 'enzyme-to-json';
import { BrowserRouter } from 'react-router-dom';

// Local JS Imports
import { BlogPost } from '.././blogpost';
import { BlogDate } from '.././components/blogdate';
import { BlogHeaderImage } from '.././components/blogheaderimage';
import { BlogParagraph } from '.././components/blogparagraph';
import { BlogPlayerCard } from '.././components/blogplayercard';
import { BlogPostContent } from '.././components/blogpostcontent';
import { BlogPostHeadline } from '.././components/blogpostheadline';
import { BlogPullQuote } from '.././components/blogpullquote';
import { BlogSnippets } from '.././components/blogsnippets';

import store from '../../store/store';

// Configurations
Enzyme.configure({ adapter: new Adapter() })

// ===========
// Suppression of SSL Warning 
//
// Code adapted from comment from sketchbuch on GitHub issues post 'Warning: useLayoutEffect does nothing on the server #1373' on 12 Oct 2019
// https://github.com/reduxjs/react-redux/issues/1373
// Accessed 09 Sep 2020
//
// __Note__:
//	
//		- Jest produces "Warning: useLayoutEffect does nothing on the server" & 'Warning: React.createElement: type is invalid -- expected a string (for built-in components)' - warning which confuses test cases 
// 		- Wrapper test suppresses that warning for clear test reports
// ===========

describe('Blog Post Tests', () => {
  	const originalConsoleError = console.error;

  	beforeEach(() => {
    	console.error = jest.fn((msg) => {
	      	if (msg.includes('Warning: useLayoutEffect does nothing on the server') || msg.includes('Warning: React.createElement: type is invalid -- expected a string (for built-in components)')) {
	       		return null;
	      	} else {
	        	originalConsoleError(msg);
	      	}
    	});
  	});

  afterEach(() => {
    console.error = originalConsoleError;
  });


// End of referenced code

  	// ===========
	// TEST 1. Check that BlogPost renders correctly
	// ===========

	describe('BlogPost', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><BlogPost/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 2. Check that BlogDate renders correctly
	// ===========

	describe('BlogDate', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><BlogDate/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 3. Check that BlogHeaderImage renders correctly
	// ===========

	describe('BlogHeaderImage', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><BlogHeaderImage/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 4. Check that BlogParagraph renders correctly
	// ===========

	describe('BlogParagraph', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><BlogParagraph/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 5. Check that BlogPlayerCard renders correctly
	// ===========

	describe('BlogPlayerCard', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><BlogPlayerCard/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 6. Check that BlogPostContent renders correctly
	// ===========

	describe('BlogPostContent', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><BlogPostContent/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 7. Check that BlogPostHeadline renders correctly
	// ===========

	describe('BlogPostHeadline', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<Provider store={store}><BlogPostHeadline/></Provider>);
			
			//
			// Simulate Component
		    const wrapper = shallow(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 8. Check that BlogPullQuote renders correctly
	// ===========

	describe('BlogPullQuote', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><BlogPullQuote/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})



	// ===========
	// TEST 9. Check that BlogSnippets renders correctly
	// ===========

	describe('BlogSnippets', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><BlogSnippets/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


});
