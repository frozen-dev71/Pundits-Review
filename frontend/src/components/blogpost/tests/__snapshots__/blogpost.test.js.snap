// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog Post Tests BlogDate should render correctly 1`] = `
<div
  class="blog-date-wrapper"
>
  <p
    class="blog-date-text body-text-light"
  />
</div>
`;

exports[`Blog Post Tests BlogHeaderImage should render correctly 1`] = `
<span
  class="blog-header-image"
  style="background-image:url(undefined)"
/>
`;

exports[`Blog Post Tests BlogParagraph should render correctly 1`] = `
<p
  class="blog-paragraph-text body-text-light"
/>
`;

exports[`Blog Post Tests BlogPlayerCard should render correctly 1`] = `<span />`;

exports[`Blog Post Tests BlogPost should render correctly 1`] = `
<div
  class="blog-post-content-wrapper"
>
  <p
    class="blog-headline-text body-text-light"
  />
  <span
    class="blog-header-image"
    style="background-image:url(undefined)"
  />
  <div
    class="blog-date-wrapper"
  >
    <p
      class="blog-date-text body-text-light"
    />
  </div>
  <span
    class="blog-list-dot"
  />
  <p
    class="blog-paragraph-text body-text-light"
  />
  <span />
  <p
    class="blog-paragraph-text body-text-light"
  />
  <p
    class="blog-pull-quote-text body-text-light"
  />
  <p
    class="blog-paragraph-text body-text-light"
  />
  <span />
  <p
    class="blog-paragraph-text body-text-light"
  />
  <p
    class="blog-pull-quote-text body-text-light"
  />
  <p
    class="blog-paragraph-text body-text-light"
  />
  <p
    class="blog-pull-quote-text body-text-light"
  />
  <p
    class="blog-paragraph-text body-text-light"
  />
  <p
    class="blog-paragraph-text body-text-light"
  />
  <p
    class="blog-paragraph-text body-text-light"
  />
  <div
    class="footer-wrapper"
  >
    <p
      class="footer-title body-text-light"
    >
      GET IN TOUCH
    </p>
    <div
      class="footer-logo-wrapper"
    >
      <a
        class="react-link"
        href="https://twitter.com/AndyClarkeMedia"
        target="_blank"
      >
        <i
          aria-hidden="true"
          class="fa fa-twitter media-icon"
        />
      </a>
      <a
        class="footer-logo heading-text react-link"
        href="/"
      >
        PR
      </a>
      <a
        class="react-link"
        href="mailto:clarkeAJ3@cardiff.ac.uk"
        target="_blank"
      >
        <i
          aria-hidden="true"
          class="fa fa-envelope media-icon"
        />
      </a>
    </div>
    <p
      class="footer-copyright-notice body-text-light"
    >
      © Andy Clarke, 2020
    </p>
  </div>
</div>
`;

exports[`Blog Post Tests BlogPostContent should render correctly 1`] = `
<div
  class="blog-post-content-wrapper"
>
  <p
    class="blog-headline-text body-text-light"
  />
  <span
    class="blog-header-image"
    style="background-image:url(undefined)"
  />
  <div
    class="blog-date-wrapper"
  >
    <p
      class="blog-date-text body-text-light"
    />
  </div>
  <span
    class="blog-list-dot"
  />
  <p
    class="blog-paragraph-text body-text-light"
  />
  <span />
  <p
    class="blog-paragraph-text body-text-light"
  />
  <p
    class="blog-pull-quote-text body-text-light"
  />
  <p
    class="blog-paragraph-text body-text-light"
  />
  <span />
  <p
    class="blog-paragraph-text body-text-light"
  />
  <p
    class="blog-pull-quote-text body-text-light"
  />
  <p
    class="blog-paragraph-text body-text-light"
  />
  <p
    class="blog-pull-quote-text body-text-light"
  />
  <p
    class="blog-paragraph-text body-text-light"
  />
  <p
    class="blog-paragraph-text body-text-light"
  />
  <p
    class="blog-paragraph-text body-text-light"
  />
  <div
    class="footer-wrapper"
  >
    <p
      class="footer-title body-text-light"
    >
      GET IN TOUCH
    </p>
    <div
      class="footer-logo-wrapper"
    >
      <a
        class="react-link"
        href="https://twitter.com/AndyClarkeMedia"
        target="_blank"
      >
        <i
          aria-hidden="true"
          class="fa fa-twitter media-icon"
        />
      </a>
      <a
        class="footer-logo heading-text react-link"
        href="/"
      >
        PR
      </a>
      <a
        class="react-link"
        href="mailto:clarkeAJ3@cardiff.ac.uk"
        target="_blank"
      >
        <i
          aria-hidden="true"
          class="fa fa-envelope media-icon"
        />
      </a>
    </div>
    <p
      class="footer-copyright-notice body-text-light"
    >
      © Andy Clarke, 2020
    </p>
  </div>
</div>
`;

exports[`Blog Post Tests BlogPostHeadline should render correctly 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Component />
</ContextProvider>
`;

exports[`Blog Post Tests BlogPullQuote should render correctly 1`] = `
<p
  class="blog-pull-quote-text body-text-light"
/>
`;

exports[`Blog Post Tests BlogSnippets should render correctly 1`] = `<span />`;
